pkg/server/instrumentation_service_test.go

```diff
@@ -3,19 +3,19 @@ package server
 import (
 	"context"
 	"io"
+	"log/slog"
 	"net/http"
 	"testing"
 	"time"
 
 	"github.com/grafana/dskit/services"
-	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestRunInstrumentationService(t *testing.T) {
-	s, err := NewInstrumentationService(log.New("test-logger"))
+	s, err := NewInstrumentationService(slog.Default().With("logger", "test-logger"))
```

Comment:
nit: can we suppress logs while testing?
```suggestion
	s, err := NewInstrumentationService(slog.NewTextHandler(io.Discard, nil))
```

Reply:
IMO logs are desirable during tests, so I don't think discarding them is a great idea.

Reply:
I agree, I thought this would always log to the console. I just checked out the code and it doesn't, only when you pass `-v` or the test fails.